runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -9317,6 +9317,379 @@ static TR::Register* inlineStringHashCode(TR::Node* node, bool isCompressed, TR:
    return hash;
    }
 
+TR::Register* J9::X86::TreeEvaluator::inlineVectorizedHashCode(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node *fromIndexNode = node->getChild(1);
+   TR::Node *initialValueNode = node->getChild(3);
+   TR::Node *elementTypeNode = node->getChild(4);
+   TR::Register* registerHash = NULL;
+
+   if (!(fromIndexNode->getOpCodeValue() == TR::iconst && fromIndexNode->getInt() == 0))
+      return NULL; // only supporting offset of const 0
+
+   switch (elementTypeNode->getConstValue())
+      {
+      case 4:  // T_BOOLEAN
+         registerHash = vectorizedHashCodeHelper(node, TR::Int8, initialValueNode, false, cg);
+         break;
+      case 8:  // T_BYTE
+         registerHash = vectorizedHashCodeHelper(node, TR::Int8, initialValueNode, true, cg);
+         break;
+      case 5:  // T_CHAR
+         registerHash = vectorizedHashCodeHelper(node, TR::Int16, initialValueNode, false, cg);
+         break;
+      case 9:  // T_SHORT
+         registerHash = vectorizedHashCodeHelper(node, TR::Int16, initialValueNode, true, cg);
+         break;
+      case 10: // T_INT
+         registerHash = vectorizedHashCodeHelper(node, TR::Int32, initialValueNode, true, cg);
+         break;
+      default:
+         return NULL;
+      }
+
+   if (registerHash != NULL)
+      cg->decReferenceCount(elementTypeNode);
+
+   node->setRegister(registerHash);
+
+   return registerHash;
+   }
+
+TR::Register *
+J9::X86::TreeEvaluator::vectorizedHashCodeReductionHelper(TR::Node* node, TR::Register **vectorRegisters, int32_t numVectors, TR::Register *tmpVectorReg, TR::Register *result, TR::VectorLength vl, TR::DataType dt, TR::CodeGenerator* cg)
```

Comment:
For consistency with your other code and readability, can you name this register `tmpVectorRegXMM` ?